lib/jsonapi/rails/configuration.rb

```diff
@@ -22,15 +22,21 @@ class Configuration < ActiveSupport::InheritableOptions; end
 
     DEFAULT_JSONAPI_EXPOSE = lambda {
       { url_helpers: ::Rails.application.routes.url_helpers }
-    }.freeze
+    }
+
+    DEFAULT_JSONAPI_FIELDS = ->() { params[:fields] }
```

Comment:
I'm afraid this won't work out of the box.

`params[:fields]` would return `AC::Parameters`, which is not compatible with `Hash` in Rails 5. So we can use something like `params[:fields]&.to_unsafe_hash` instead. But this won't work either because fields for every resource specified still need to be parsed (for instance, a request to `/api/articles?fields[posts]=title,summary&fields[comments]=content,rating` will result in a `:fields` hash `{ "posts" => "title,summary", "comments" => "content,rating" }`, which will cause an error if passed to the renderer).

So I think that it will be safe to disable parsing/processing of JSONAPI fields by default (set it to `nil` or an empty hash) and provide a parser just like you did with `jsonapi_include`.

Reply:
Updated the PR according to your feedback. At the moment I simply provide an example in the initializer but I agree it might make sense in the future to:
1. provide `Hash[fields_param.map { |k, v| [k.to_sym, v.split(',').map!(&:to_sym)] }]` as a method somewhere,
2. expose some mechanism to parse the `fields` param by default without necessarily using it (i.e. make it into a hash and turn it back to an `AC::Parameters` instance that the user can directly validate).

Reply:
Cool, I like it. I can test this PR (fields part) in an hour on my website.

Reply:
Awesome, thanks!

Reply:
Tested `jsonapi_fields` configuration, works without any issue.